signer/src/bitcoin/client.rs

```diff
@@ -63,11 +63,17 @@ impl BitcoinInteract for ApiFallbackClient<BitcoinCoreClient> {
     }
 
     async fn estimate_fee_rate(&self) -> Result<f64, Error> {
-        todo!() // TODO(542)
+        // TODO(542): This function is supposed to incorporate other fee
+        // estimation methods, in particular the ones in the
+        // src/bitcoin/fees.rs module.
+        self.exec(|client, _| async { client.estimate_fee_rate(1) })
+            .await
+            .map(|feerate| feerate.sats_per_vbyte)
     }
```

Comment:
qq: (also for `broadcast_transaction`) most of the times the impl for `ApiFallbackClient<C>` is just a passthrough to `C`, why here (and in other places) we are implementing "business logic" instead?

Reply:
Done in https://github.com/stacks-network/sbtc/pull/676/commits/9d6f7c14bab26983cc5db4d3192b6dc6653e8b91